src/backend/distributed/transaction/transaction_management.c

```diff
@@ -205,8 +205,17 @@ InCoordinatedTransaction(void)
 void
 Use2PCForCoordinatedTransaction(void)
 {
-	Assert(InCoordinatedTransaction());
-
+	/*
+	 * If this transaction is also a coordinated
+	 * transaction, use 2PC. Otherwise, this
+	 * state change does nothing.
+	 *
+	 * In other words, when this flag is set,
+	 * we "should" use 2PC when needed (e.g.,
+	 * we are in a coordinated transaction and
+	 * the coordinated transaction does a remote
+	 * modification).
+	 */
```

Comment:
It is kind of unusual that we decide to use 2PC while processing a non-remote command, but still doesn't feel wrong. I think the comment and the name of `ShouldCoordinatedTransactionUse2PC` reflects this good enough.

I'm open for other suggestions if you have any